src/helics/core/coreTypeOperations.cpp

```diff
@@ -162,45 +164,45 @@ namespace core {
     }
 
 #ifndef ENABLE_ZMQ_CORE
-#    define ZMQ_AVAILABILITY false
+    static bool constexpr zmq_availability{false};
 #else
-#    define ZMQ_AVAILABILITY true
+    static bool constexpr zmq_availability{true};
 #endif
 
 #ifndef ENABLE_MPI_CORE
-#    define MPI_AVAILABILITY false
+    static bool constexpr mpi_availability{false};
 #else
-#    define MPI_AVAILABILITY true
+    static bool constexpr mpi_availability{false};
```

Comment:
```suggestion
    static bool constexpr mpi_availability{true};
```

Reply:
clang-tidy recognizing that the shared_ptr was not necessary in that context and did nothing except an extra increment.